knowledge_base/Web-Wallets.md

```diff
@@ -0,0 +1,123 @@
+
+1. The user begins the authentication flow by clicking  the `Sign Up` or `Create Account` buttons in the header, or else by attempting to perform an auth-gated action.
+
+2. Assuming a user selects the `Create Account` button, the [CreateAccountModal](../packages/commonwealth/client/scripts/views/modals/AuthModal/CreateAccountModal/CreateAccountModal.tsx) is opened.
+
+    1. The user is shown 2 options, `Create a wallet` and `I have a wallet`.
+
+        + If `Create a wallet` is selected, only SSO options are rendered.
+
+        + If `I have a wallet` is selected, only Web3 wallets are rendered.
+
+    2. The user selects from the displayed wallet or SSO options.
+
+        + If the user already has an account linked to the selected wallet or SSO option, then they will be signed into the app once the auth process is complete.
+
+        + If the user does not have an account linked to the selected wallet or SSO option, the [WelcomeOnboardModal](../packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/WelcomeOnboardModal.tsx) is opened once the authentication process completes.
+
+3. Assuming a user clicked on the `Signup` button, the [SignInModal](../packages/commonwealth/client/scripts/views/modals/AuthModal/SignInModal/SignInModal.tsx) is opened.
+
+    + The user can select from any available wallet or SSO option.
+
+    + If the user already has an account linked to the selected wallet or SSO option, they are signed into the app once the authentication process completes.
+
+4. If the user does not have an account linked to the selected wallet or SSO option, the [`AuthTypeGuidanceModal`](../packages/commonwealth/client/scripts/views/modals/AuthModal/AuthTypeGuidanceModal/AuthTypeGuidanceModal.tsx) is opened.
+
+    + If `Create an account` is selected, the `CreateAccountModal` is displayed.
+
+    + If `Sign in another way` is selected, the `SignInModal` is displayed.
+
+```mermaid
+flowchart LR
+    AuthTypeGuidanceModal --> CreateAccountModal
+    AuthTypeGuidanceModal --> SignInModal
+
+    CreateAccountModal --> WelcomeOnboardModal
+```
+
+### Sign-in state
+
+A user's active account is stored in `app.user.activeAccount`, and updated in the [login controller](../packages/commonwealth/client/scripts/controllers/app/login.ts).
+
+### Flowchart
+
+```mermaid
+flowchart LR
+    NewSession --> SignIn
+    NewSession --> CreateAccount
+
+    SignIn --> KnownSignInInfo
+    SignIn --> UnknownSignInInfo
+    KnownSignInInfo --> SignInFlow
+    UnknownSignInInfo --> CreateAccount
+
+    CreateAccount --> ExistingWallet
+    CreateAccount --> NewWallet
+
+    ExistingWallet --> Username
+    NewWallet --> Username
+
+    Username --> TypeName
+    Username --> GenerateName
+    Username --> ImportSSOName
+
+    TypeName --> CustomizeExperience
+    GenerateName --> CustomizeExperience
+    ImportSSOName --> CustomizeExperience
+
+    CustomizeExperience --> JoinCommunity
```

Comment:
Is this for the user-onboarding flow?

If yes, this would be more like
```diff
- ExistingWallet --> Username
- NewWallet --> Username
+ ExistingWallet --> WalletsList # only wallet options are displayed
+ NewWallet --> SSOList  # only sso options are displayed 
```

when user selects and option from sso/wallet list, an account is created, then we show the welcome modal and other steps like `PersonalInformation`, `CustomizeExperience`, `JoinCommunity`